gsrs-module-substances-core/src/main/java/ix/ginas/utils/validation/validators/TagsValidator.java

```diff
@@ -0,0 +1,159 @@
+package ix.ginas.utils.validation.validators;
+
+import gsrs.module.substance.repository.SubstanceRepository;
+import ix.core.validator.GinasProcessingMessage;
+import ix.core.validator.ValidatorCallback;
+import ix.ginas.models.v1.Substance;
+import ix.ginas.utils.validation.AbstractValidatorPlugin;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+
+import java.util.*;
+@Slf4j
+
+/*
+
+=== Tags Validator Documentation ===
+
+The idea of substance tags appears in two ways in the GSRS. 1) in the list of names. 2) in the list of tags.
+
+A tag is included in a substance name by putting a bracketed term at the end of the name.
+For example:
+  ASPIRIN [INN]
+When this name is added, the GSRS will extract "INN" and index this value for faceting.
+The facet category called "Source Tag" works with this indexed value.
+There is ALSO a substance->tags list of Keyword objects.
+In the GSRS Frontend, tag values can be added manually to this list.
+Thus, there are two sources of truth for substance tags.
+This validator may be used to keep the lists consistent.
+
+This should be configured in your gsrs-ci gsrs-ci/substances/src/main/resources/application.conf.
+
+  # Without this setting, the validator will not be run on submission of a substance.
+  gsrs.validators.substances += {
+                "validatorClass" = "ix.ginas.utils.validation.validators.TagsValidator",
+                "newObjClass" = "ix.ginas.models.v1.Substance"
+        }
+
+  # These may be added, though defaults are set in the code.
+  gsrs.substance.addtagwhenintagspresentinnames=false
+  gsrs.substance.removetagwhenintagsmissingfromnames=false
+
+*** Note also that there is an idea of the locator.  This is used in conjunction with references.
+Currently this is OFF in GSRS (controlled by a boolean value extractLocators). If that were on,
+a method in NamesValidator.java (addLocator) would add tags found in names to the tags list. We should
+probably remove this piggybacking in case locators is ever turned back on.
+*/
+
+
+public class TagsValidator extends AbstractValidatorPlugin<Substance> {
+
+    @Value("#{new Boolean('${gsrs.substance.addtagwhenintagspresentinnames:false}')}")
```

Comment:
You want this to be not part of the global config variables, but part of THIS validator's config initialization. @ChemMitch does this for 

https://github.com/ncats/gsrs-spring-module-substances/blob/b2514e01f1f2e5de80bbdc24c1ee562a6a493384/gsrs-module-substances-core/src/main/java/ix/ginas/utils/validation/validators/CodeUniquenessValidator.java#L66

And a few others, and it's pretty simple. You just need your Validator to be JSON serializable/deserializable. It'll be deserialized by:

https://github.com/ncats/gsrs-spring-starter/blob/7d1b4324ab200ad7743caf32b7ee71d70bd3de0e/gsrs-spring-boot-autoconfigure/src/main/java/gsrs/validator/DefaultValidatorConfig.java#L57

So, if you add `@Data` to your Validator (or make sure there are setters for each config param), it'll "just work" when you specify your validator in the config file with extra stuff. You won't need the `@Value` annotation anymore.

Also, use the java primative boolean instead of the boxed Boolean.